packages/test-runner-commands/src/sendMousePlugin.ts

```diff
@@ -0,0 +1,156 @@
+import { TestRunnerPlugin } from '@web/test-runner-core';
+import type { ChromeLauncher } from '@web/test-runner-chrome';
+import type { PlaywrightLauncher } from '@web/test-runner-playwright';
+import type { WebdriverLauncher } from '@web/test-runner-webdriver';
+
+type MousePosition = [number, number];
+type MouseButton = 'left' | 'middle' | 'right';
+
+export type SendMousePayload = MovePayload | ClickPayload | DownPayload | UpPayload;
+
+export type MovePayload = {
+  type: 'move';
+  position: MousePosition;
+};
+
+export type ClickPayload = {
+  type: 'click';
+  position: MousePosition;
+  button?: MouseButton;
+};
+
+export type DownPayload = {
+  type: 'down';
+  button?: MouseButton;
+};
+
+export type UpPayload = {
+  type: 'up';
+  button?: MouseButton;
+};
+
+function isObject(payload: unknown): payload is Record<string, unknown> {
+  return payload != null && typeof payload === 'object';
+}
+
+function isSendMousePayload(payload: unknown): payload is SendMousePayload {
+  const validTypes = ['move', 'click', 'down', 'up'];
+  const validButtons = ['left', 'middle', 'right'];
+
+  if (!isObject(payload)) {
+    throw new Error('You must provide a `SendMousePayload` object');
+  }
+
+  if (typeof payload.type !== 'string' || !validTypes.includes(payload.type)) {
+    throw new Error(
+      `You must provide a type option with one of the following values: ${validTypes.join(', ')}.`,
+    );
+  }
+
+  if (['click', 'move'].includes(payload.type)) {
+    if (
+      !Array.isArray(payload.position) ||
+      typeof payload.position[0] !== 'number' ||
+      typeof payload.position[1] !== 'number' ||
+      !/^\d+$/.test(payload.position.join(''))
```

Comment:
Consider using [`Number.isInteger`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) for better readability.

Reply:
Done.